### DISCUSSION TITLE
Nuget namespaces for aspire component.

### DISCUSSION THREAD

**Discussion by tomohisa** [microsoft/aspire#1543] [Q&A] [answered]
## Our situation.

Hi, we are really excited for the Aspire project!

We have nuget package that **uses Cosmos DB** and **can be used in app services**.
It is Event Sourcing and CQRS Framework that uses Cosmos DB as data store.
https://www.nuget.org/packages/Sekiban.Infrastructure.Cosmos

We would like to make **simple aspire componen**t for using cosmos db added with Aspire.

```csharp
builder.AddAzureCosmosDB("SekibanCosmos");
```

Our component code will be simple one, get keyed cosmos client component from DI, and passed it to our sekiban framework.

And we would like to start using in our test sample projects. So we would like to release our preview nuget package for our component that uses aspire cosmos db.

## How to name our own components in Nuget package?

I realized nuget namespace for Aspire.* is reserved.

Thus should we make preview nuget package as

`Sekiban.Infrastructure.Cosmos.Aspire`

instead for

`Aspire.Sekiban.Infrastructure.Cosmos`

I looked if you have an official documentation, but I could not find it.
If there is official way to add "Aspire.*" namespaces, let us know. 

But we are totally fine for adding nuget package name under our own namespaces like `Sekiban.Infrastructure.Cosmos.Aspire`

**Comment by davidfowl** [comment]
cc @eerhardt

**Comment by eerhardt** [comment, marked as answer]
From a technical perspective, there isn't anything special about Aspire Component packages - at least nothing you can't do yourself. They don't depend on "Aspire-specific" things, but instead just depend on existing concepts like:

* Microsoft.Extensions.Hosting.Abstractions
* OpenTelemetry (for example, OpenTelemetry.Extensions.Hosting)
* HealthChecks (if necessary)

In your situation, I would think you wouldn't even need to create a new NuGet package, but instead you could just add this `builder.AddAzureCosmosDB("SekibanCosmos");` extension method to your existing `Sekiban.Infrastructure.Cosmos` NuGet package.

To answer the general question, yes the `Aspire.*` namespace is reserved for "official" Aspire Components (i.e. ones owned by the [Aspire NuGet account](https://www.nuget.org/profiles/aspire)). If you were looking to create one of your own, I would suggest putting it in your own namespace - like `Sekiban.Infrastructure.Cosmos.Aspire` or `Sekiban.Aspire.Infrastructure.Cosmos`.

cc @DamianEdwards

**Reply by tomohisa** [reply]
Thank you so much! We would do like you said. If we wanted to separate package, follow what you suggest! Thanks for taking time to answer! Appreciate all of your great work.
